endpoints/api/namespacequota.py

```diff
@@ -75,252 +74,322 @@ class OrganizationQuota(ApiResource):
+        quota_limit_threshold = quota_limit_data["threshold_percent"]
+
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+
+        quota_limit = model.namespacequota.get_namespace_quota_limit_list(
+            quota,
+            quota_type=quota_type,
+            percent_of_limit=quota_limit_threshold,
         )
 
-        if quota is not None:
-            msg = "quota limit already exists"
+        if quota_limit:
+            msg = "Quota limit already exists"
             raise request_error(message=msg)
 
-        reject_quota = model.namespacequota.get_namespace_reject_limit(namespace)
-        if reject_quota is not None and model.namespacequota.is_reject_limit_type(
-            quota_limit_data["quota_type_id"]
-        ):
-            msg = "You can only have one Reject type of quota limit"
-            raise request_error(message=msg)
+        if quota_limit_data["type"].lower() == "reject" and quota_limit:
+            raise request_error(message="Only one quota limit of type 'Reject' allowed.")
 
         try:
-            model.namespacequota.create_namespace_limit(
-                orgname=namespace,
-                percent_of_limit=quota_limit_data["percent_of_limit"],
-                quota_type_id=quota_limit_data["quota_type_id"],
+            model.namespacequota.create_namespace_quota_limit(
+                quota,
+                quota_type,
+                quota_limit_threshold,
             )
             return "Created", 201
         except model.DataModelException as ex:
             raise request_error(exception=ex)
 
-    @nickname("changeOrganizationQuotaLimit")
-    @validate_json_request("NewOrgQuotaLimit")
-    def put(self, namespace):
 
-        superperm = SuperUserPermission()
+@resource("/v1/organization/<orgname>/quota/<quota_id>/limit/<limit_id>")
+@show_if(features.QUOTA_MANAGEMENT)
+class OrganizationQuotaLimit(ApiResource):
+    schemas = {
+        "UpdateOrgQuotaLimit": {
+            "type": "object",
+            "description": "Description of changing organization quota limit",
+            "properties": {
+                "type": {
+                    "type": "string",
+                    "description": 'Type of quota limit: "Warning" or "Reject"',
+                },
+                "threshold_percent": {
+                    "type": "integer",
+                    "description": "Quota threshold, in percent of quota",
+                },
+            },
+        },
+    }
 
-        if not superperm.can():
+    @nickname("getOrganizationQuotaLimit")
+    def get(self, orgname, quota_id, limit_id):
+        orgperm = OrganizationMemberPermission(orgname)
+        if not orgperm.can():
             raise Unauthorized()
 
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        quota_limit = model.namespacequota.get_namespace_quota_limit(quota, limit_id)
```

Comment:
this will cause 3 db calls if not more. Recommend doing this in one possibly 2 trips. This pattern is also used throughout this class.